tck/faces23/ajax/src/test/java/ee/jakarta/tck/faces/test/javaee8/ajax_selenium/Spec1423IT.java

```diff
@@ -0,0 +1,123 @@
+ */
+
+package ee.jakarta.tck.faces.test.javaee8.ajax_selenium;
+
+import ee.jakarta.tck.faces.test.util.selenium.BaseITNG;
+import ee.jakarta.tck.faces.test.util.selenium.WebPage;
+import jakarta.faces.component.UIViewRoot;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.time.Duration;
+
+import static org.junit.Assert.assertTrue;
+
+public class Spec1423IT extends BaseITNG {
+
+    /**
+     * @see UIViewRoot#addComponentResource(jakarta.faces.context.FacesContext, jakarta.faces.component.UIComponent, String)
+     * @see https://github.com/jakartaee/faces/issues/1423
+     */
+    @Test
+    public void testSpec1423Basic() throws Exception {
+        WebPage page = getPage("spec1423.xhtml");
+        WebElement button;
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().isEmpty());
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form1:addViaHead"));
+        button.click();
+        page.waitReqJs(Duration.ofMillis(6000));
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+    }
+
+    @Test
+    public void testSpec1423() throws Exception {
+        WebPage page = getPage("spec1423.xhtml");
+        WebElement button;
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().isEmpty());
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form1:addViaHead"));
+        button.click();
+
+        page.waitForCondition(wd -> {
+            return page.findElement(By.id("stylesheetResult")).getText().trim().isEmpty() &&
+                    page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead");
+        });
+
+
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaHead"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().isEmpty());
+
+        button = page.findElement(By.id("form2:addViaInclude"));
+        button.click();
+
+        page.waitForCondition(wd -> {
+            return page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(255, 0, 0)") &&
+                    page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaInclude");
+        });
+
+        button = page.findElement(By.id("form1:addViaBody"));
+        button.click();
+        page.waitReqJs(Duration.ofMillis(6000));
+        assertTrue(page.findElement(By.id("scriptResult")).getText().trim().equals("addedViaBody"));
+        assertTrue(page.findElement(By.id("stylesheetResult")).getText().trim().equals("rgb(255, 0, 0)"));
+
+        button = page.findElement(By.id("form2:addViaInclude"));
```

Comment:
It's probably better if page was reloaded so that the script/style results of previous test step are guaranteed to be cleared out.

Reply:
yes that would mean a complete test rewrite, also this would miss the double include check. (Aaka if a script already is loaded it should not be executed a second time)
This test is rather complicated in a sense that it tests for several things at once.

Reply:
btw generally it would make sense to replace the waits with waitForCondition in the other tests as well, but for this one it was the most pressing.